src/Features/LanguageServer/Protocol/Handler/IRequestHandler.cs

```diff
@@ -29,4 +29,10 @@ internal interface IRequestHandler<RequestType, ResponseType> : IRequestHandler
         /// <returns>the LSP response.</returns>
         Task<ResponseType> HandleRequestAsync(RequestType request, ClientCapabilities clientCapabilities, string? clientName, CancellationToken cancellationToken);
     }
+
+    internal enum RequestProcessingMode
+    {
+        Serial,
+        Parallel
+    }
```

Comment:
Do we need a `None` that doesn't use the queue at all? That would match current behaviour, but don't know if it would be meaningfully useful over `Parallel`

Reply:
i would leave it simple

Reply:
Consecutive/Concurrent?

Reply:
+1 serial/consecutive and parallel/concurrent seem like the only valid options, so we should force the handlers to choose.